### DISCUSSION TITLE
Tips : Do not use RDP, but rustdesk with MFA and local relay

### DISCUSSION THREAD

**Discussion by nliaudat** [LoicVeirman/HardenAD#137] [Ideas]
_Tiering is good, but tiering with bastion is better._ 

Actually, securing RDP is very hard : 

- using Restricted Admin mode = you must manage the machine rights to access ressources
- using the standard mode = you let ntlm or kerberos traces on the clients workstation

My solution is to forgive RDP and use Rustkesk with a local relay + MFA authentication. 

A rustdesk local relay can be installed in 15 minutes with docker and nothing leave the vlan you designed. 
With complex static password + MFA, the access to the PAWS is much more secured than with RDP and leave no credentials.

That low cost solution simplify the bation/tiering paradigm, which is complex to implement with no security breach. 

With strict firewall rules (with group or user authentication), you can jump from L2-paws to L0-paws while remaining well insulated and without leaving traces.

**Comment by LoicVeirman** [comment]
I personnaly recommand to my clients Wallix, Systencia or even a specific configuration with Royal Server / Royal TS (this one protect identity). When clients are fully managing virtualized boxes through Hyper-V, I disable RDP and use VMConnect through a specific delegation model, then authenticating with SCL (no user pass, that obvious). This however require a bastion.

I was not aware of rustdesk, I'll get a look to it. What make me in concern with it however will be the docker under the hood: that is another risks surface that should be analyzed.

**Reply by nliaudat** [reply]
Rustkesk use end to end ChaCha20-Poly1305 algo. The communication pass encryted in the relay server. 
Chacha20-Poly1305 is similar to AES-256 in term of security. 

The default port used (21116,21117,21118,21119) can be reduced and changed. I mean some good firewall policies are also needed to have a secured solution. 

Rustkesk relay server can run on any windows and linux distro but I prefer using [Photon OS](https://vmware.github.io/photon/) to run the docker.

The main goal is to kick RDP out off the security equation cause it's hard to secure (VPN, NLA, FIPS encryption level, remove UDP channel) and is the target of choise for hackers since 1996.

**Comment by LoicVeirman** [comment]
Rustdesk will not be implemented as part of this project. However, it could added to the community tool as a separate project that could work separately or over a HAD deployment. Let me know if you want to manage it.
